_includes/flipflop2.html

```diff
@@ -1,359 +1,501 @@
+        $("#confirmStep2").click(function () {
+          var inp = $("#step2 div.form-input").find("input");
+          for (var i = 0; i < inp.length; i++) {
+            if (inp[i].value == "") {
+              inp[i].focus();
+              flag = i;
+              break;
+            } else {
+              //alert(inp[i].id + ":" + inp[i].value);
+              flipflops[inp[i].id] = inp[i].value;
+            }
+          }
+
+          if (flag != null) {
+            inp[flag].focus();
+            flag = null;
+          } else {
+            var summary =
+              "<p><strong>Summary of User Input.</strong>&nbsp;&nbsp;A sequential circuit that has <strong>" +
+              flipflopCount +
+              "</strong> <strong>" +
+              input["flipflop"] +
+              "</strong> flipflops( ";
+
+            for (var i = 0; i < flipflopCount; i++) {
+              summary += String.fromCharCode(65 + i) + " ";
+            }
+
+            summary += "), <strong>" + inputCount + "</strong> inputs( ";
+
+            for (var i = 0; i < inputCount; i++) {
+              summary += String.fromCharCode(97 + 25 - inputCount + i) + " ";
+            }
+
+            summary += " )";
+
+            if (outputCount == 1) {
+              summary += " and an output(Z).";
+            } else {
+              summary += ".";
+            }
+
+            summary +=
+              " The flip-flop input functions and the circuit function are as follows:</p>";
+            summary += '<div class="form-input">';
+
+            var keys = Object.keys(flipflops);
+
+            for (var i = 0; i < keys.length; i++) {
+              summary +=
+                "<h3>" + keys[i] + " = " + flipflops[keys[i]] + "</h3> ";
+            }
+
+            summary += "</div>";
+
+            $("#summary #summa").append(summary);
+            $("#step2").hide();
+            $("#summary").fadeIn();
+          }
+        });
+
+        $("#backToStep2").click(function () {
+          $("#summary #summa").empty();
+          $("#summary").hide();
+          $("#step2").fadeIn();
+        });
+
+        $("#restart").click(function () {
+          $("#step1 select").val("0");
+          $("#step2 div.form-input").empty();
+          $("#summary #summa").empty();
+          $("#table_container").empty();
+          $("#stateDesc").empty();
+          $("#stateTable").hide();
+          $("#about").fadeIn();
+        });
+
+        $("#confirmSummary").click(function () {
+          var output = [];
+          output = generateStateTable(
+            flipflopCount,
+            inputCount,
+            flipflops,
+            input["flipflop"]
+          );
+
+          var table = "<p><table class='table'><thead>";
+
+          // //table += "<th colspan=" + flipflopCount + ">Present State</th><th colspan=" + inputCount + ">Input</th><th colspan=" + flipflopCount + ">Next State</th><th colspan=" + flipflops.length + ">FlipFlop Inputs</th>";
```

Comment:
consider removing it ?

Reply:
Was commented by someone else. Hence didn't remove. Should I?

Reply:
@b30wulffz I think you should